src/display.cpp

```diff
@@ -233,7 +256,13 @@ display::display(const display_context * dc, std::weak_ptr<wb::manager> wb, repo
 
 	set_idle_anim_rate(preferences::idle_anim_rate());
 
-	zoom_index_ = std::distance(zoom_levels.begin(), std::find(zoom_levels.begin(), zoom_levels.end(), zoom_));
+	unsigned int tile_size = preferences::tile_size();
+	if(tile_size < MinZoom || tile_size > MaxZoom)
+		tile_size = DefaultZoom;
+	zoom_index_ = get_zoom_levels_index(tile_size);
+	zoom_ = zoom_levels[zoom_index_];
```

Comment:
I wonder if you should set `last_zoom_ = zoom_ / 2` here, to match how `zoom_` and `last_zoom_` are initialized?

Reply:
As discussed in #wesnoth-dev, I think we should keep the existing behaviour of the default zoom key. It sounded like you agreed.